bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -175,7 +125,7 @@ public ObjectId() {
      * @param date the date
      */
     public ObjectId(final Date date) {
-        this(dateToTimestampSeconds(date), MACHINE_IDENTIFIER, PROCESS_IDENTIFIER, NEXT_COUNTER.getAndIncrement(), false);
+        this(dateToTimestampSeconds(date), NEXT_COUNTER.getAndIncrement(), false);
```

Comment:
Does this go against the spec? 

> When the counter overflows (i.e., hits 16777215+1), the counter MUST be reset to 0.

If so, what are the implications?  Why doesn't this cause a problem now?

Reply:
I have changed the constructor to check for counter overflow and reset the counter if overflow occurs.